benchmarks/persistance/s3_persistance.py

```diff
@@ -0,0 +1,89 @@
+"""Classes for persisting and loading experiment results to a S3-Bucket."""
+
+import json
+import os
+
+import boto3
+import boto3.session
+from attr import define, field
+from attrs.validators import instance_of
+from git import Repo
+
+from benchmarks import Benchmark, Result
+
+
+@define
+class S3ExperimentResultPersistence:
+    """Class for persisting experiment results in an S3 bucket."""
+
+    _bucket_name: str = field(validator=instance_of(str), init=False)
+    """The name of the S3 bucket where the results are stored."""
+
+    _object_session = boto3.session.Session()
+    """The boto3 session object. This will load the respective credentials
+    from the environment variables within the container."""
+
+    _branch: str = field(validator=instance_of(str), init=False)
+    """The branch of the Baybe library from which the workflow was started."""
+
+    _workflow_run_id: str = field(validator=instance_of(str), init=False)
+    """The ID of the workflow run."""
+
+    @_bucket_name.default
+    def _default_bucket_name(self) -> str:
+        ENVIRONMENT_NOT_SET = "BAYBE_PERFORMANCE_PERSISTENCE_PATH" not in os.environ
+        if ENVIRONMENT_NOT_SET:
+            raise ValueError(
+                "The environment variable "
+                "BAYBE_PERFORMANCE_PERSISTENCE_PATH is not set. "
+                "A bucket name must be provided."
+            )
+        return os.environ["BAYBE_PERFORMANCE_PERSISTENCE_PATH"]
+
+    @_branch.default
+    def _default_branch(self) -> str:
+        repo = Repo(search_parent_directories=True)
+        current_branch = repo.active_branch.name
+        sanitized_branch = current_branch.replace("/", "-")
```

Comment:
Why the sanitation? Does this character cause troubles?

Reply:
Yes, git allows '/' as being part of the branches name but for s3, this will be interpreted as a folder separator.